api_v2/src/leaves/leaves.service.ts

```
@@ -161,6 +173,361 @@ export class LeavesService {
+          notification.type === 'leave_resolved' ||
+          notification.type === 'undertime_resolved';
+
+        const statusIsApproved =
+          parsedData.Status === RequestStatusEnum.APPROVED;
+
+        if (!isApprovedNotification || !statusIsApproved) {
+          // This project leader hasn't approved yet
+          return false;
+        }
+      } catch (e) {
+        // If we can't parse, assume not approved
+        return false;
+      }
+    }
+
+    // All project leaders have approved
+    return true;
+  }
+
+  /**
+   * Checks if any project leader has disapproved the leave
+   * @param leaveId - The leave ID to check
+   * @returns true if any project leader has disapproved
+   */
+  private async hasAnyProjectLeaderDisapproved(
+    leaveId: number,
+  ): Promise<boolean> {
+    // Get all project leaders assigned to this leave
+    const projectLeaders = await this.prisma.projectLeadApprover.findMany({
+      where: {
+        leaveId,
+      },
+      select: {
+        projectLeaderId: true,
+      },
+    });
+
+    const projectLeaderIds = [
+      ...new Set(
+        projectLeaders
+          .map((pl) => pl.projectLeaderId)
+          .filter((id): id is number => id !== null),
+      ),
+    ];
+
+    // Check notifications for disapprovals
+    for (const leaderId of projectLeaderIds) {
+      const notification = await this.prisma.notification.findFirst({
+        where: {
+          relatedEntityId: leaveId,
+          recipientId: leaderId,
+        },
+        orderBy: {
+          createdAt: 'desc',
+        },
+      });
+
+      if (notification) {
+        try {
+          const parsedData =
+            typeof notification.data === 'string'
+              ? JSON.parse(notification.data)
+              : notification.data;
+
+          if (parsedData.Status === RequestStatusEnum.DISAPPROVED) {
+            return true;
+          }
+        } catch (e) {
+          // Continue checking other leaders
```

Add error message

The code swallows errors silently, making debugging impossible if something fails (like malformed JSON or a database issue).

- Silent error swallowing in notification update methods: catch (e) { continue }
- Makes debugging difficult and hides potential issues
- Recommendation: Log errors or use proper error handling